src/Provider/Youtube/Format.php

```diff
@@ -128,14 +128,36 @@ private function parseUrl()
 
         // The video signature need to be deciphered
         if (isset($this->raw_data['s']) and $this->config['decipher_signature']) {
+
             // TODO: Remove signature decipher from Format
-            $player_info = SignatureDecipher::getPlayerInfoByVideoId($this->getVideoId());
+            $videoIds = $this->getCache()->get('videoIds', null);
+            if(!$videoIds) $videoIds = [];
+            
+            if(!in_array($this->getVideoId(), $videoIds)) {
+                $videoIds[] = $this->getVideoId();
+                $this->getCache()->set('videoIds', $videoIds, 3600*24);
+                
+                // getPlayerInfoByVideoId should be run once only for a video
+                $player_info = SignatureDecipher::getPlayerInfoByVideoId($this->getVideoId());
+
+                $playerID = $player_info[0];
+                $playerURL = $player_info[1];
+                $this->getCache()->set('playerID', [$playerID, $playerURL], 3600*24);
```

Comment:
Please cache directly $player_info and remove line 143 and 144.

`$this->getCache()->set('playerID', $player_info, 3600*24);`